packages/cypress/src/add-commands/login.ts

```diff
@@ -125,26 +122,146 @@ function loginByForm(commandOptions: CommandLoginOptions) {
       userCredentials.email,
       commandOptions.entryPointUriPath,
     ];
-    // const mcUrl = new URL(appConfig.mcApiUrl);
-    // const mcFrontendHostname = mcUrl.hostname.replace('mc-api', 'mc');
 
+    /**
+     * The function is used by Cypress `session` command to store the
+     * browser state after executing the authentication flow.
+     */
     function authCallback() {
-      cy.visit(url, { onBeforeLoad: commandOptions.onBeforeLoad });
-
-      // NOTE: using `cy.origin` is currently disabled as it does not seem to properly work.
-      // Interestingly, starting an application locally using Vite works, however not when using Webpack.
-      // For now we keep it disabled until we find a solution.
-
-      // // https://cypress.io/blog/2022/04/25/cypress-9-6-0-easily-test-multi-domain-workflows-with-cy-origin/
-      // cy.origin(
-      //   mcFrontendHostname,
-      //   { args: userCredentials },
-      //   fillLoginForm
-      // );
-      fillLoginForm(userCredentials);
-
-      // Wait for the route to be loaded so that the session can be saved.
-      cy.url().should('include', url);
+      // Get the feature flags from the Merchant Center API
+      cy.request({
+        method: 'POST',
+        url: `${appConfig.mcApiUrl}/graphql`,
+        body: {
+          operationName: 'AllFeatures',
+          query: `query AllFeatures { allFeatures { name value reason } }`,
+        },
+        headers: {
+          'content-type': 'application/json',
+          'x-graphql-target': GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND,
+          'x-graphql-operation-name': 'AllFeatures',
+          'x-project-key': projectKey,
+        },
+      }).then(
+        (res: {
+          body: {
+            data: {
+              allFeatures: { name: string; value: string; reason: string }[];
+            };
+          };
+        }) => {
+          const enableIdentity = res.body.data.allFeatures.find(
+            (feature) => feature.name === 'enableGlobalIdentity'
+          );
+          const isGlobalIdentityEnabled = Boolean(enableIdentity?.value);
+
+          // Visit the application URL, which triggers then the login flow.
+          cy.visit(url, { onBeforeLoad: commandOptions.onBeforeLoad });
+
+          // If Identity is enabled, we need to login using the Identity API.
+          if (isGlobalIdentityEnabled) {
+            const identityUrl =
+              Cypress.env('IDENTITY_URL') ||
+              'https://identity.commercetools.com';
+
+            // eslint-disable-next-line cypress/no-unnecessary-waiting
+            cy.wait(1000);
```

Comment:
I haven't found a better way to do this. If we don't wait, Cypress is confused and errors out because the redirect to Identity hasn't happened yet.
By waiting, we give time to do the redirect and pick up the new origin.

Reply:
@CarlosCortizasCT For reference, this is the error we see if we don't use `wait`

<img width="432" height="549" alt="image" src="https://github.com/user-attachments/assets/7a99254f-8697-47de-a9b8-978ca7fb91ea" />